lib/gui/app/models/store.js

```diff
@@ -550,3 +508,34 @@ module.exports = _.merge(redux.createStore(storeReducer, DEFAULT_STATE), {
   Actions: ACTIONS,
   Defaults: DEFAULT_STATE
 })
+
+/**
+ * @summary Observe the store for changes
+ * @param {Function} onChange - change handler
+ * @returns {Function} unsubscribe
+ * @example
+ * store.observe((newState) => {
+ *   // ...
+ * })
+ */
+module.exports.observe = (onChange) => {
+  let currentState = null
+
+  /**
+   * @summary Internal change detection handler
+   * @private
+   * @example
+   * store.subscribe(changeHandler)
+   */
+  const changeHandler = () => {
+    const nextState = module.exports.getState()
+    if (!_.isEqual(nextState, currentState)) {
+      currentState = nextState
+      onChange(currentState)
+    }
+  }
+
+  const unsubscribe = module.exports.subscribe(changeHandler)
+  changeHandler()
+  return unsubscribe
```

Comment:
Do we need the `changeHandler()` between `const unsubscribe` and `return unsubscribe` or could we consolidate the const-setting with the return statement? Either works for me I'm just curious if I've missed something subtle.